_posts/2020-02-11-spring-boot-openapi.md

```diff
@@ -0,0 +1,394 @@
+* [OpenAPI (Swagger) Editor](https://marketplace.visualstudio.com/items?itemName=42Crunch.vscode-openapi), which adds rich support (IntelliSense, linting, schema enforcement, etc...) for OpenAPI development.
+* [openapi-lint](https://marketplace.visualstudio.com/items?itemName=mermade.openapi-lint), a linter for OpenAPI.
+
+In this tutorial, we'll be using VSCode, therefore I have installed [OpenAPI (Swagger) Editor](https://marketplace.visualstudio.com/items?itemName=42Crunch.vscode-openapi).
+
+## Creating an API Spec with the Swagger Editor
+
+Let's create our own OpenAPI specification in a YAML document. To make it easier to follow, we'll split the discussion into separate parts of the YAML document we're creating.
+
+If you want to learn more details about the OpenAPI-Specification you can visit the [Github repository](https://github.com/OAI/OpenAPI-Specification/blob/master/vtersions/3.0.2.md#versions).
+
+
+### General Information
+
+We start off with some general information about our API at the top of our document:
+
+```yaml
+openapi: 3.0.2
+info:
+  title: Reflectoring
+  description: "Tutorials on Spring Boot and Java, thoughts about the Software Craft, and relevant book reviews. Because it's just as important to understand the Why as it is to understand the How. Have fun!"
+  termsOfService: http://swagger.io/terms/
+  contact:
+    email: petros.stergioulas94@gmail.com
+  license:
+    name: Apache 2.0
+    url: http://www.apache.org/licenses/LICENSE-2.0.html
+  version: 0.0.1-SNAPSHOT
+externalDocs:
+  description: Find out more about Reflectoring
+  url: https://reflectoring.io/about/
+servers:
+- url: https://reflectoring.swagger.io/v2
+```
+
+The [`openapi`](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#openapi-object) field allows us to define the version of the OpenAPI spec that our document follows.
+
+Within the [`info`](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#infoObject) section, we add some information about our API. The fields should be pretty self-explanatory.
+
+Finally, in the [`servers`](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#serverObject) section, we provide a list of servers that implement the API.
+
+### Tags
+
+Then comes some additional metadata about our API.
+
+```yaml
+tags:
+- name: user
+  description: Operations about user
+  externalDocs:
+    description: Find out more about our store
+    url: http://swagger.io
+```
+
+* The [`tags`](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md#fixed-fields) section, provide additional metadata, which can be used to make our API more readable and easier to follow. Each tag should be unique.
+
+### Paths
+
+Here we will describe some paths, to make it clean how we can create our own. But, what are paths? Relative paths holds the information  to the individual endpoints and their operations.
+
+```yaml
```

Comment:
I removed one of the endpoints because I think one is enough to show how it works and the reader will be less distracted this way.